### ISSUE TITLE
window.fullscreen(true) broken in 1.4.26

### ISSUE THREAD

**Issue by dividebysandwich** [fltk-rs/fltk-rs#1534]
When switching from 1.4.25 to 1.4.26 I noticed that under Windows, wind.fullscreen(true) no longer makes the window go fullscreen. It retains the window decorations. With 1.4.25 this worked fine. This happens in both single-screen and dual-screen setups.

EDIT: I'm actually a bit surprised that this seems to now be a persistent issue. I tried `cargo update -p fltk --precise 1.4.25` as well as `cargo update -p fltk-sys --precise 1.4.25` but I still get the broken fullscreen problem. I also tried cargo clean, removing rust and reinstalling, the works.

**Comment by MoAlyousef** [comment]
Yeah I'm able to reproduce the issue. I think it's an upstream one. I'll try to repro then report to the fltk devs.

**Comment by dividebysandwich** [comment]
Yeah looks like you are spot-on. What irks me a bit is that I don't know how to get back to a working version, it seems once you have the newer version the precompiled binary just sticks somewhere...

**Comment by MoAlyousef** [comment]
If you're using ```
[patch.crates-io]
fltk = { git = "https://github.com/fltk-rs/fltk-rs" }
```
from the previous issue, you'll have to remove it. Then you can pin version 1.4.25 with `fltk = "=1.4.25"` in your Cargo.toml.

**Comment by MoAlyousef** [comment]
Yeah I was bisecting and came across this:
https://github.com/fltk/fltk/commit/130f5b6dc1c2258b1a05c9319d796b96dc2d4f7e

This will be required for activating fullscreen on FLTK.

**Comment by dividebysandwich** [comment]
Ok but there's still a problem: When updating the window with new frame contents and redrawing, it flips out of fullscreen. Then, I have to do fullscreen(false) followed by fullscreen(true) to get it back to fullscreen.

**Comment by MoAlyousef** [comment]
Can you share a minimal repro?
I can't seem to reproduce this behavior.

**Comment by dividebysandwich** [comment]
That is really difficult because it's actually a multi-step problem that requires user input to show. But I think I found the problem and it's in my code.

Basically, I am loading all kinds of images, including animated GIFs. I found that when loading an animated GIF on windows, the built in canvas resize works on the first loaded image, but causes problems with the positioning when switching to a new image(the image gets stuck halfway past the top left corner). In my ignorance I fixed this by setting the window size to the full screen size again. This worked fine with fltk 1.4.25.

Now with 1.4.26 properly requiring a resizable window to be able to make it fullscreen, this set_size() call on what is already a fullscreen window wreaks all kinds of havoc, and to get out of that you have to get out of fullscreen, and then set fullscreen(true) again.

I fixed this by removing the set_size call, and instead passing AnimGifImageFlags::DONT_RESIZE_CANVAS to the load() function.

TLDR: My workarounds (which worked in 1.4.25) caused the problem, and the current fltk behavior is actually more consistent anyway. I'm closing this issue because as you pointed out, setting the window to be resizable is now required to make it fullscreen.

Thanks for the speedy responses btw!
